### PR TITLE
Add aws.metrics_names_fingerprint field and mark it as a dimension

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
## Proposed commit message

This PR adds `aws.metrics_names_fingerprint` which is a hash of the list of metric names exist in each document. For metrics coming in from Firehose, there can be cases two documents have the same timestamp, dimension, namespace, accountID, exportARN and region BUT from two different requests. With adding the `aws.metrics_names_fingerprint` as TSDB dimension, we are able to ingest these two documents into ES.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Relates https://github.com/elastic/integrations/pull/11239

### CODE REVIEW

File: packages/aws/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "2.24.2"
+  changes:
+    - description: Add aws.metrics_names_fingerprint.
+      type: enhancement
```
Comment (efd6) [inline packages/aws/changelog.yml:5]: Enhancement or bug? If enhancement, please bump the minor rather than the patch.

Comment (kaiyan-sheng) [reply packages/aws/changelog.yml:5]: well i have mix feeling about it 😂 I will make it a bug fix so we can release for 2.24.2!

**Review by efd6 — APPROVED** [review]
Approving as codeowner.

**Review by axw — APPROVED** [review]

**Review by agithomas — APPROVED** [review]
LGTM!

**Pull request merged**